pkg/window/strategy/sliding/sliding.go

```diff
@@ -0,0 +1,183 @@
+// Package sliding implements Sliding windows. Sliding windows are defined by a static window size
+// e.g. minutely windows or hourly windows and a fixed "slide". This is the duration by which the boundaries
+// of the windows move once every <slide> duration.
+// Package sliding also maintains the state of active windows.
+// Watermark is used to trigger the expiration of windows.
+package sliding
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/window"
+	"github.com/numaproj/numaflow/pkg/window/keyed"
+)
+
+// Sliding implements sliding windows
+type Sliding struct {
+	// Length is the duration of the window
+	Length time.Duration
+	// offset between successive windows.
+	// successive windows are phased out by this duration.
+	Slide time.Duration
+	// entries is the list of active windows that are currently being tracked.
+	// windows are sorted in chronological order with the earliest window at the head of the list.
+	// list.List is implemented as a doubly linked list which allows us to traverse the nodes in
+	// both the directions.
+	// Although the worst case time complexity is O(n), because of the time based ordering and
+	// since the elements are rarely out of order, the amortized complexity works out to be closer to O(1)
+	// Because most of the keys are expected to be associated with the most recent window, we always start
+	// the traversal from the tail of the list for Get and Create Operations. For Remove Operations, since
+	// the earlier windows are expected to be closed before the more recent ones, we start the traversal
+	// from the Head.
+	entries *list.List
+	lock    sync.RWMutex
+}
+
+var _ window.Windower = (*Sliding)(nil)
+
+// NewSliding returns a Sliding windower
+func NewSliding(length time.Duration, slide time.Duration) *Sliding {
+	return &Sliding{
+		Length:  length,
+		Slide:   slide,
+		entries: list.New(),
+		lock:    sync.RWMutex{},
+	}
+}
+
+// AssignWindow returns a set of windows that contain the element based on event time
+func (s *Sliding) AssignWindow(eventTime time.Time) []window.AlignedKeyedWindower {
```

Comment:
Verified with https://github.com/numaproj/numaflow/pull/157#issuecomment-1247067394, it looks right :)